packages/ckeditor5-ui/src/autocomplete/autocompleteview.ts

```diff
@@ -0,0 +1,232 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/autocomplete/autocompleteview
+*/
+
+import { getOptimalPosition, type PositioningFunction, type Locale, global, toUnit, Rect } from '@ckeditor/ckeditor5-utils';
+import SearchTextView, { type SearchTextViewConfig } from '../search/text/searchtextview';
+import type SearchResultsView from '../search/searchresultsview';
+import type InputBase from '../input/inputbase';
+import type { FilteredViewExecuteEvent } from '../search/filteredview';
+
+import '../../theme/components/autocomplete/autocomplete.css';
+
+/**
+ * The autocomplete component's view class. It extends the {@link module:ui/search/text/searchtextview~SearchTextView} class
+ * with a floating {@link #resultsView} that shows up when the user starts typing and hides when they blur
+ * the component.
+ */
+export default class AutocompleteView<
+	TQueryFieldView extends InputBase<HTMLInputElement | HTMLTextAreaElement>
+> extends SearchTextView<TQueryFieldView> {
+	/**
+	 * The configuration of the autocomplete view.
+	 */
+	protected override _config: AutocompleteViewConfig<TQueryFieldView>;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale: Locale, config: AutocompleteViewConfig<TQueryFieldView> ) {
+		super( locale, config );
+
+		this._config = config;
+
+		const toPx = toUnit( 'px' );
+
+		this.extendTemplate( {
+			attributes: {
+				class: [ 'ck-autocomplete' ]
+			}
+		} );
+
+		const resultsView = this.resultsView as AutocompleteResultsView;
```

Comment:
Add `declare public resultsView: AutocompleteResultsView;` to this class and get rid of those casts. The reason is to make it explicit that `this.resultsView` in this class always conforms to `AutocompleteResultsView` instead of assuring the compiler with the implicit knowledge at every use site with casts.